### PR TITLE
Feature/592 recipient info

### LINKED ISSUE #592: Swap Screen: recipient info
For: 0x... (x) / (Connected with incorrect wallet) (hover should show tooltip with whole address)
For: You (ok) / (Connected with correct wallet)
For: Anyone

Screenshot 2022-08-02 at 21 34 59
Screenshot 2022-08-02 at 21 35 06
Screenshot 2022-08-02 at 21 35 13

### PR DESCRIPTION
**Pull request by piersss**
Fixes #592 

This PR will introduce all possible combinations of the Take flow header components. `OrderDetailWidgetHeader` uses the following props:

```
orderStatus: OrderStatus;
orderType: OrderType;
recipientAddress?: string;
transactionLink?: string;
userAddress?: string;
```

OrderStatus can be `canceled`, `open` and `taken`.
OrderType can be `public-listed`, `public-unlisted` and `private`.

After this I will do the same for `ActionButtons` underneath the `SwapInputs`. Same way as SwapWidget is setup.

Screenshot 2022-08-13 at 11 34 24

### CODE REVIEW

File: src/components/ExpiryIndicator/ExpiryIndicator.tsx [hunk 1]
```diff
@@ -8,14 +8,16 @@ import {
...
 type ExpiryIndicatorProps = {
   expiry: Date;
+  className?: string;
 };
 
-const ExpiryIndicator: FC<ExpiryIndicatorProps> = ({ expiry }) => {
+const ExpiryIndicator: FC<ExpiryIndicatorProps> = ({ expiry, className }) => {
   const { t } = useTranslation();
 
   const hasPassed = useMemo(
```
Comment (coltraneyan) [inline src/components/ExpiryIndicator/ExpiryIndicator.tsx:23]: ```hasExpired``` could be more clear

File: src/components/ExpiryIndicator/ExpiryIndicator.tsx [hunk 2]
```diff
@@ -28,9 +30,9 @@ const ExpiryIndicator: FC<ExpiryIndicatorProps> = ({ expiry }) => {
...
 
   return (
-    <Container>
+    <Container className={className}>
       <StyledPill>
-        {hasPassed ? t("common.expired") : t("common.expiresIn")}
+        <Text>{hasPassed ? t("common.expired") : t("common.expiresIn")}</Text>
         {!hasPassed && <Strong>{timeLeft}</Strong>}
       </StyledPill>
       <StyledTooltip>{format(expiry, "dd-MMM-yyyy pppp")}</StyledTooltip>
```
Comment (coltraneyan) [inline src/components/ExpiryIndicator/ExpiryIndicator.tsx:38]: ```dd-MMM-yyyy``` a typo?

**Review by coltraneyan — COMMENTED** [review]
Looks good!

File: src/components/ExpiryIndicator/ExpiryIndicator.tsx (hunk 1 above)
Comment (piersss) [reply src/components/ExpiryIndicator/ExpiryIndicator.tsx:23]: Expiry is a `Date`. It's used in other types like `UnsignedOrder` and `Order`, so it's best to keep things consistent.

File: src/components/ExpiryIndicator/ExpiryIndicator.tsx (hunk 2 above)
Comment (piersss) [reply src/components/ExpiryIndicator/ExpiryIndicator.tsx:38]: date-fns weird formatting rules 😜 https://date-fns.org/v2.29.1/docs/format

File: src/components/ExpiryIndicator/ExpiryIndicator.tsx (hunk 1 above)
Comment (coltraneyan) [reply src/components/ExpiryIndicator/ExpiryIndicator.tsx:23]: ```hasPassed``` -> ```hasExpired``` was what i meant, my bad.

Comment (piersss) [reply src/components/ExpiryIndicator/ExpiryIndicator.tsx]: Fixed!

**Review by coltraneyan — APPROVED** [review]

**Pull request merged**